### ISSUE TITLE
Support ES Modules

### ISSUE THREAD

**Issue by dyoder** [StableLib/stablelib#41]
I'm adding this per @dchest's comment in #28:

> One "feature" that I'd like to add in the future is to support the modern module system in addition to CommonJS, so I'll work on it some time in the future. It will probably require reworking of the random generator detection.

FWIW, we need this for our work (we also need this for Tweet-NaCl). If there are no immediate plans to do this, we can submit a PR.

**Comment by dyoder** [comment]
@dchest I have a working model for doing this, but I don't want to apply across all the packages unless it's acceptable to you.

- Basically, I add a second tsconfig file, like [this](https://github.com/dashkite/stablelib/commit/5da75092ba7ea590ec1b46e56d2647db3008c79c#diff-8af94ee7770e2005e8b99086d3e8e0e3e6c97acbe8fa119fb9ff1faa4cf108b7).
- Then I add exports and alter the build command like [this](https://github.com/dashkite/stablelib/commit/5da75092ba7ea590ec1b46e56d2647db3008c79c#diff-fb1d7c20ebd0e78dc04cddb73a6600135b61cbe390f3859bb48a46b94605c19a).
- Running `npm run build` now will build both versions, placing the esm-compatible version into `lib-es6`.

I've tested this with `base64` and `utf8` and the tests still pass, although that's unsurprising since we still build into the `lib` directory as before. I didn't attempt to get Jasmine to run the tests in `lib-es6` as well, but I can do that if this approach makes sense to you.

**Comment by josh-i386g** [comment]
Hi, is this related to the errors I'm getting?

On  eslint I use `"plugin:@typescript-eslint/recommended-type-checked"` instead of `"plugin:@typescript-eslint/recommended"`.

```js
import * as base64 from "@stablelib/base64";
import * as utf8 from "@stablelib/utf8";
import * as hex from "@stablelib/hex";
import * as nacl from "@stablelib/nacl";
```

```
Unsafe call of an `any` typed value.eslint[@typescript-eslint/no-unsafe-call](https://typescript-eslint.io/rules/no-unsafe-call)
Unsafe argument of type `any` assigned to a parameter of type `Uint8Array`.eslint[@typescript-eslint/no-unsafe-argument](https://typescript-eslint.io/rules/no-unsafe-argument)
Unsafe member access .decode on an `any` value.eslint[@typescript-eslint/no-unsafe-member-access](https://typescript-eslint.io/rules/no-unsafe-member-access)
function decode(s: string): Uint8Array
```

I have to use the following:

```
/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
```

Just curious if this is something on stablelib's types, thank you.

**Comment by PhearZero** [comment]
Hello @dchest and @dyoder. Hope you are well, didn't know about stable lib until today. Just submitted a draft PR to Tweet-NaCl for ESM. Would love to sync up and help in any way.

- https://github.com/dchest/tweetnacl-js/pull/261

**Comment by dchest** [comment]
Fixed in 2.0.
